TestPidListerFailure test fails on cgroupsv2

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->


**What happened**:

```console
$ make test WHAT="./pkg/util/oom"
+++ [0210 00:04:55] Running tests without code coverage
--- FAIL: TestPidListerFailure (0.00s)
    oom_linux_test.go:103: 
        	Error Trace:	oom_linux_test.go:103
        	Error:      	Should be true
        	Test:       	TestPidListerFailure
        	Messages:   	expected getPids to return not exists error. Got open /sys/fs/cgroup/does/not/exist/cgroup.procs: no such file or directory
FAIL
FAIL	k8s.io/kubernetes/pkg/util/oom	0.027s
FAIL
make: *** [Makefile:184: test] Error 1
```

**What you expected to happen**: The test should pass

**How to reproduce it (as minimally and precisely as possible)**:

I think the key here is being on a linux host with cgroupsv2 enabled.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:


/sig node

--- a/pkg/util/oom/oom_linux_test.go
+++ b/pkg/util/oom/oom_linux_test.go
@@ -19,6 +19,8 @@ limitations under the License.
 package oom
 
 import (
+	"os"
+
 	"github.com/opencontainers/runc/libcontainer/cgroups"
 	"testing"
 
@@ -100,5 +102,5 @@ func TestOOMScoreAdjContainer(t *testing.T) {
 
 func TestPidListerFailure(t *testing.T) {
 	_, err := getPids("/does/not/exist")
-	assert.True(t, cgroups.IsNotFound(err), "expected getPids to return not exists error. Got %v", err)
+	assert.True(t, cgroups.IsNotFound(err) || os.IsNotExist(err), "expected getPids to return not exists error. Got %v", err)
 }

